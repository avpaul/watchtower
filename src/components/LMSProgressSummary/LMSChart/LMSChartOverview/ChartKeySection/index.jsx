import React from 'react';

const CharKeySection = () => (
  <div className="chart-keys">
    <div className="chart-keys__item">
      <div className="chart-keys__item--circle green" />
      <span>Scores 2 or more</span>
    </div>
    <div className="chart-keys__item">
      <div className="chart-keys__item--circle red" />
      <span>Scores less than 2</span>
    </div>
    <div className="chart-keys__item">
      <div className="chart-keys__item--circle orange" />
      <span>No submission</span>
    </div>
    <div className="chart-keys__item">
      <div className="chart-keys__item--circle grey" />
      <span>Not reviewed</span>
    </div>
  </div>
);

export default CharKeySection;
