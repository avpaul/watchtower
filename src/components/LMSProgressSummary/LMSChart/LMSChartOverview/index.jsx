import React from 'react';
import PropTypes from 'prop-types';
import ChartKeySection from './ChartKeySection';
/**
 *
 * @param {number, number, number} props
 *
 * @returns {JSX} React JSX
 */
const LMSChartOverview = ({
  numOfOutputsGreaterThanOne,
  numOfOutputsSuggested,
  numOfTotalOutputs
}) => (
  <div className="lms-chart-overview">
    <div className="output-score-summary">
      <div className="output-score-summary__progress">
        <div>
          <strong>Outputs with scores 2+ </strong> -{' '}
          {numOfOutputsGreaterThanOne}/{numOfOutputsSuggested}
        </div>
        <div>
          <strong>Required </strong> - {numOfOutputsSuggested}/
          {numOfTotalOutputs}
        </div>
      </div>
    </div>
    <ChartKeySection />
  </div>
);

LMSChartOverview.propTypes = {
  numOfOutputsSuggested: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  numOfTotalOutputs: PropTypes.number.isRequired,
  numOfOutputsGreaterThanOne: PropTypes.number.isRequired
};

export default LMSChartOverview;
