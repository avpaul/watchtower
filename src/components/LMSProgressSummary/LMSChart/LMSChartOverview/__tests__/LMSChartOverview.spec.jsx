/* eslint no-param-reassign: 0 */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import LMSChartOverview from '..';

describe('LMSChartOverview component', () => {
  const props = {
    numOfOutputsGreaterThanOne: 3,
    numOfOutputsSuggested: 9,
    numOfOutputsSubmitted: 15
  };
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<LMSChartOverview {...props} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LMSChartOverview {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders chart overview details', () => {
    expect(
      wrapper.find('.output-score-summary__progress').length
    ).toBeDefined();
    expect(wrapper.find('.chart-keys').length).toBeDefined();
  });
});
