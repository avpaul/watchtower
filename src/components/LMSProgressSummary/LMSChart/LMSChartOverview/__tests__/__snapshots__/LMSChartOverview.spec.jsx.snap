// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LMSChartOverview component matches snapshot 1`] = `
<div
  className="lms-chart-overview"
>
  <div
    className="output-score-summary"
  >
    <div
      className="output-score-summary__progress"
    >
      <div>
        <strong>
          Outputs with scores 2+ 
        </strong>
         -
         
        3
        /
        15
      </div>
      <div>
        <strong>
          Required 
        </strong>
         - 
        15
        /
        9
      </div>
    </div>
  </div>
  <CharKeySection />
</div>
`;
